Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift

```diff
@@ -401,6 +416,96 @@ class SentryProfilerSwiftTests: XCTestCase {
             options.profilesSampler = { _ in return -0.01 }
         }
     }
+
+    /// based on ``SentryTracerTests.testFinish_WithoutHub_DoesntCaptureTransaction``
+    func testProfilerCleanedUpAfterTransactionDiscarded_NoHub() throws {
```

Comment:
`m`: Excellent test cases. I think we should still add the following test cases to cover more edge cases, which you may adapt, of course, or skip cause they might be covered with existing test cases:

1. Starting a tracer / transaction, never calling finish, and the the tracer is deallocated
2. Starting multiple tracers / transactions at the same time to ensure the counting logic in SentryProfiledTracerConcurrency works properly and all transactions get their data from the same profile
3. Starting a manual transaction, keeping it running for over 30 seconds, which is the profiling timeout. Start a second transaction after the profile is timed out so a new one starts. Then, finish the second transaction to ensure it gets the second profile. Finally, finish the first transaction to ensure it gets the first profile.

Reply:
We do cover (3) already [here](https://github.com/getsentry/sentry-cocoa/blob/afd1a08398bee1bb289679eefe2793e635b3c915/Tests/SentryProfilerTests/SentryProfilerSwiftTests.swift#L288), but the other two are worth exploring. 

And for clarification, for (1), and what you're discussing [here](https://github.com/getsentry/sentry-cocoa/pull/3154#discussion_r1267857820), this is a case where a tracer is started, but the handle we return is not retained or goes out of scope, and so is deallocated?

Thanks for spec'ing these out for me, this is helpful for me to understand all the exit points of the tracer, which are many!

Reply:
Added tests for 1 and 2 in the latest commit. for 2, we already had a concurrent tracer scenario, so I just needed to add the appropriate assertions to it.

Reply:
> And for clarification, for (1), and what you're discussing https://github.com/getsentry/sentry-cocoa/pull/3154#discussion_r1267857820, this is a case where a tracer is started, but the handle we return is not retained or goes out of scope, and so is deallocated?

Yes, the handle will be deallocated as soon as it goes out of scope.

It's a crazily complicated topic. I'm looking forward to keeping the profiler just running with a ring buffer. That would make things easier. (Last final words 😅 )

Reply:
Oh yeah!